#919, #881: Fixed 3rd part URLs-related issues

kittaakos opened the pull request:
### Motivation
<!-- Why this pull request? -->

This PR fixes various 3rd party URLs issues:
 - fixes a problem when parsing an empty/blank string  into 3rd part URLs array (#919),
 - the height of the 3rd party URLs `textarea` height (#881)

### Change description
<!-- What does your code do? -->

 - Single-sourced and fixed the way we parse a string to 3rd party URLs, added tests.
 - Adjusted the style of the dialog content `display` to fix the `textarea` height in the 3rd party URLs selector dialog.

Before:
<img width="720" alt="Screen Shot 2022-03-17 at 14 18 00" src="https://user-images.githubusercontent.com/1405703/158818111-33e5981e-e385-4dec-bb02-52ba9235e202.png">

After:
<img width="716" alt="Screen Shot 2022-03-17 at 14 17 47" src="https://user-images.githubusercontent.com/1405703/158818089-b4f7ce1e-ef46-4f7d-8319-8f2a165767cb.png">


### Other information
<!-- Any additional information that could help the review process -->

I am happy to split the PR into two, however, I thought it's easier to verify the 3rd party URLs changes in one shot. Let me know if a PR split is desired.

> [ ] Docs have been added / updated (for bug fixes / features)

I do not know what this means. Thanks!

### Reviewer checklist

* [ ] PR addresses a single concern. (no, I can open a follow-up with #881 only, if required)
* [x] The PR has no duplicates (please search among the [Pull Requests](https://github.com/arduino/arduino-ide/pulls) before creating one)
* [x] PR title and description are properly filled.
* [ ] Docs have been added / updated (for bug fixes / features)

fstasi reviewed (approved):
lgtm

fstasi commented:
Thanks @kittaakos for contributing, works like a charm to me.
@per1234 do you want to give this PR a try?

per1234 reviewed (changes requested):
The **File > Preferences > Additional Boards Manager URLs** field no longer allows me to type a comma at the end of the line. This is problematic because it is a comma-separated list so a user who wants to add multiple URLs directly via the field will expect to be able to add a URL, then a comma, then the next URL.

1. Select **File > Preferences...** from the Arduino IDE menus.
1. Type a URL into the "**Additional Boards Manager URLs**" field.
1. Type `,`

🐛 The expected `,` character is not added. If you don't notice then the next URL will not be separated from the previous one.

It does allow commas to be added within existing text. It is only ignored when adding one at the end of the line.

ubidefeo commented:
@kittaakos please address changes requested by @per1234 so I can approve this

kittaakos commented:
Thanks for the note. I did not forget about this. I will try to find some time to update the PR this week, @ubidefeo. The bug reported by Per is valid. I most likely need to change the `<input>` from a controlled React component to an uncontrolled one. To sum up, the fix is slightly bigger than I have expected.

kittaakos commented:
I have updated this PR. This time, I spent more time verifying my changes locally. Hopefully, I did not overlook anything and fixed the issue. Please review. Thank you!

per1234 reviewed (approved):
Working perfectly for me now. Thanks Akos!
